manta-crypto/src/arkworks/algebra.rs

```diff
@@ -104,3 +104,24 @@ where
         Self(scalar, PhantomData)
     }
 }
+
+/// Returns the power-of-two table of size `powers` such that entry at index `i` is `base * 2^i`.
+#[inline]
+pub fn precomputed_bases_custom<C>(base: C, powers: usize) -> impl Iterator<Item = C>
+where
+    C: ProjectiveCurve,
+{
+    core::iter::successors(Some(base), |base| Some(*base + *base)).take(powers)
+}
+
+/// Returns the power-of-two table of whose size is modulus bits of the constraint field of `C` such that entry at index `i` is `C::prime_subgroup_generator() * 2^i`.
+#[inline]
+pub fn precomputed_bases<C>() -> impl Iterator<Item = C>
+where
+    C: ProjectiveCurve,
+{
+    precomputed_bases_custom(
+        C::prime_subgroup_generator(),
+        <ConstraintField<C> as PrimeField>::size_in_bits(),
+    )
+}
```

Comment:
Sorry, I meant to add the concept of these tables to the `algebra` module. Something that every ECLAIR backend can use not the arkworks-specific one. These tables should be integrated into the rest of the circuit architecture. If you need help implementing this we can meet tomorrow and iterate on this or I can push some code as well.

Reply:
no worries, I will update this tomorrow

Reply:
This is also my main comment. I would add, that you use the compiler to compute `fixed_based_scalar_mul`, but not to precompute the values. Wouldn't the values in this precomputed table depend also on the choice of compiler?

Reply:
Thanks for the comment! I've updated the code to make table generation generic, so it no longer depends on the choice of compiler.